src/EntityconnectWidgetProcessor.php

```diff
@@ -0,0 +1,399 @@
+
+    // Get the parents.
+    if (isset($this->widget['#field_parents'])) {
+      foreach ($this->widget['#field_parents'] as $key1 => $parent) {
+        if (!isset($parents)) {
+          $parents = $parent;
+        }
+        else {
+          $parents .= "-" . $parent;
+        }
+      }
+    }
+    $parents = isset($parents) ? $parents : '';
+
+    $fieldStorage = $this->fieldDefinition->getFieldStorageDefinition();
+    $extraClass = isset($this->widget['#type']) ? $this->widget['#type'] : 'autocomplete';
+    $extraClass .= $fieldStorage->getCardinality() > 1 ? ' multiple-values' : ' single-value';
+    $extraClass .= (isset($this->widget['#multiple']) && $this->widget['#multiple'] == TRUE) ? ' multiple-selection' : ' single-selection';
+    if (isset($this->widget['#type'])) {
+      if ((isset($this->widget['#multiple']) && $this->widget['#multiple'] == TRUE) || $this->widget['#type'] == 'radios' || $this->widget['#type'] == 'checkboxes') {
+        $element['#attributes']['class'][] = 'inline-label';
+      }
+    }
+
+    // Set the class strings for the button
+    $buttonClasses = array(
+      'extra_class' => $extraClass,
+      'parents_class' => $parents,
+    );
+
+    // Set the correct element to attach to.
+    if ($key === 'all') {
+      // Options widget.
+      if (isset($this->widget['#type'])) {
+        $widgetElement = &$element;
+      }
+      // Autocomplete Tags widget.
+      else {
+        $widgetElement = &$element['widget'];
+      }
+    }
+    // Autocomplete widget.
+    else {
+      $widgetElement = &$element['widget'][$key];
+    }
+
+    $this->attachAddButton($widgetElement, $buttonClasses);
+    $this->attachEditButton($widgetElement, $buttonClasses, $key);
+
+  }
+
+  /**
+   * Attach the Add button.
+   *
+   * @param array $element
+   *   The widget container element.
+   * @param string $entityconnect_classes
+   *   Button CSS definition array:
+   *   - 'extra_class': extra css class string
+   *   - 'parents_class': parents class string
+   * @param string $key optional
+   *   Always 'all' for Add button.
+   */
+  protected function attachAddButton(&$element, $entityconnect_classes, $key = 'all') {
+
+    // Button values are opposite; 0=On, 1=Off.
+    $addbuttonallowed = !$this->entityconnectSettings['buttons']['button_add'];
+    $addIcon = $this->entityconnectSettings['icons']['icon_add'];
+
+    // Get the subset of target bundles the user has permission to create.
+    $acceptableTypes = array();
+    foreach ($this->acceptableTypes as $bundle) {
+      if (\Drupal::entityTypeManager()
+        ->getAccessControlHandler($this->entityType)
+        ->createAccess($bundle)
+      ) {
+        $acceptableTypes[] = $bundle;
+      }
+    }
+
+    // Now we need to make sure the user should see this button.
+    if (\Drupal::currentUser()->hasPermission('entityconnect add button') && $addbuttonallowed && $acceptableTypes) {
```

Comment:
`$acceptableTypes` cause the issue in user reference as there is no option for type in User reference.